packages/gguf/src/types.ts

```diff
@@ -66,39 +57,18 @@ interface General {
 	"general.quantization_version": number;
 }
 
-type Attention<TArchitecture extends Architecture> =
-	| { [K in `${TArchitecture}.attention.head_count`]: number }
-	| { [K in `${TArchitecture}.attention.head_count_kv`]: number }
-	| { [K in `${TArchitecture}.attention.layer_norm_epsilon`]: number }
-	| { [K in `${TArchitecture}.attention.layer_norm_rms_epsilon`]: number }
-	| { [K in `${TArchitecture}.attention.alibi_bias_max`]: number }
-	| { [K in `${TArchitecture}.attention.clip_kqv`]: number }
-	| { [K in `${TArchitecture}.attention.use_norm`]: number };
-
-type Rope<TArchitecture extends Architecture> =
-	| { [K in `${TArchitecture}.rope.dimension_count`]: number }
-	| { [K in `${TArchitecture}.rope.freq_base`]: number }
-	| { [K in `${TArchitecture}.rope.scale`]: number }
-	| { [K in `${TArchitecture}.rope.scale_linear`]: number };
-
-type ModelBase<
+export type ModelBase<
 	TArchitecture extends
 		| Architecture
 		| `encoder.${Extract<Architecture, "whisper">}`
 		| `decoder.${Extract<Architecture, "whisper">}`,
-> =
-	| { [K in `${TArchitecture}.layer_count`]: number }
-	| { [K in `${TArchitecture}.feed_forward_length`]: number }
-	| { [K in `${TArchitecture}.context_length`]: number }
-	| { [K in `${TArchitecture}.embedding_length`]: number }
-	| { [K in `${TArchitecture}.block_count`]: number };
-
-type MOE<TArchitecture extends Architecture> =
-	| { [K in `${TArchitecture}.expert_count`]: number }
-	| { [K in `${TArchitecture}.expert_used_count`]: number };
+> = { [K in `${TArchitecture}.layer_count`]: number } & { [K in `${TArchitecture}.feed_forward_length`]: number } & {
+	[K in `${TArchitecture}.context_length`]: number;
+} & { [K in `${TArchitecture}.embedding_length`]: number } & { [K in `${TArchitecture}.block_count`]: number };
```

Comment:
It's maybe more readable to have:

```suggestion
Record<
	| `${TArchitecture}.layer_count`
	| `${TArchitecture}.feed_forward_length`
	| `${TArchitecture}.context_length`
	| `${TArchitecture}.embedding_length`
	| `${TArchitecture}.block_count`,
	number
>
```

Reply:
Actually it's being formatted by `pnpm run format`. Can we maybe ignore specific lines from the formatter?

Reply:
maybe we merge this one and improve in later PRs?

Reply:
Yeah I think we can merge now. I'm not having much time for doing code atm, but I'll open another pull to resolve this and the also the proposal related to usage of these types (hopefully I'll have time this weekend).